WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/Catalog Visibility/ProductCatalogVisibilityViewController.swift

```diff
@@ -0,0 +1,234 @@
+
+    @IBOutlet weak var tableView: UITableView!
+
+    // Completion callback
+    //
+    typealias Completion = (_ productSettings: ProductSettings) -> Void
+    private let onCompletion: Completion
+
+    private var productSettings: ProductSettings
+
+    private let sections: [Section]
+
+    /// Init
+    ///
+    init(settings: ProductSettings, completion: @escaping Completion) {
+        productSettings = settings
+        sections = [Section(rows: [.featuredProduct]), Section(rows: [.visible, .catalog, .search, .hidden])]
+        onCompletion = completion
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavigationBar()
+        configureMainView()
+        configureTableView()
+        configureTableViewFooter()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        onCompletion(productSettings)
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        tableView.updateFooterHeight()
+    }
+
+}
+
+
+// MARK: - View Configuration
+//
+private extension ProductCatalogVisibilityViewController {
+
+    func configureNavigationBar() {
+        title = NSLocalizedString("Catalog Visibility", comment: "Product Catalog Visibility navigation title")
+
+        removeNavigationBackBarButtonText()
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureTableView() {
+        tableView.register(SwitchTableViewCell.loadNib(), forCellReuseIdentifier: SwitchTableViewCell.reuseIdentifier)
+        tableView.register(BasicTableViewCell.loadNib(), forCellReuseIdentifier: BasicTableViewCell.reuseIdentifier)
+        tableView.register(TwoColumnSectionHeaderView.loadNib(), forHeaderFooterViewReuseIdentifier: TwoColumnSectionHeaderView.reuseIdentifier)
+
+        tableView.dataSource = self
+        tableView.delegate = self
+
+        tableView.backgroundColor = .listBackground
+        tableView.removeLastCellSeparator()
+    }
+
+    func configureTableViewFooter() {
+        /// `tableView.tableFooterView` can't handle a footerView that uses autolayout only.
+        /// Hence the container view with a defined frame.
+        let footerContainer = UIView(frame: CGRect(x: 0, y: 0, width: Int(tableView.frame.width), height: 1))
+        let footerView = FootnoteTableFooterView.instantiateFromNib() as FootnoteTableFooterView
+        footerView.footnoteLabel.text = NSLocalizedString("This setting determines which shop pages products will be listed on.",
+                                                          comment: "Footer text in Product Catalog Visibility")
+        tableView.tableFooterView = footerContainer
+        footerContainer.addSubview(footerView)
```

Comment:
I was wondering if we could use the native footer since the design looks similar? like in the Application Log view controller:

https://github.com/woocommerce/woocommerce-ios/blob/7aa29d27a4b6fb317c1e892c1f2ec24e57c7e72e/WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApplicationLogViewController.swift#L148-L154

or are we using the custom footer view because the design is quite different from the native table footer view?

Reply:
We also have a `TableFooterView` and `UITableView`'s `updateFooterHeight` extension that is used in a few settings screens if it fits the use cases 🙂

Reply:
I have tried `TableFooterView` but it can't be used because the design is different (I also tried to customize it, but there are too many differences), but I did not know of the existence of another header (the native one) that is exactly like our design.
I discovered that the footer title follows our design only if the table view style is set to "Grouped", otherwise is too big.
Thanks for this suggestion @jaclync!